mino/minogrpc/server.go

```diff
@@ -535,19 +538,41 @@ func (o *overlay) GetCertificateStore() certs.Storage {
 	return o.certs
 }
 
-// Join sends a join request to a distant node with token generated beforehands
-// by the later.
+// Join sends a join request to a distant node with a token generated by the
+// remote node.
+// The certHash is used to make sure that no man-in-the-middle intercepts the
+// communication.
+// If the certHash is empty, it supposes that a transparent proxy is handling
+// the TLS connection and that we can trust the CAs in place.
 func (o *overlay) Join(addr *url.URL, token string, certHash []byte) error {
 
-	target := session.NewAddress(addr.Host + addr.Path)
+	host := addr.Host
+	if addr.Port() == "" {
+		switch addr.Scheme {
+		case "http":
+			host += ":80"
+		case "https":
+			host += ":443"
+		default:
+			return xerrors.Errorf("address doesn't contain a port and uses "+
+				"an unknown scheme: %s", addr.Scheme)
+		}
+	}
```

Comment:
why is this needed ? Effectively the code is adding default ports for those schemes, that's normally not needed 🤔